subprojects/docs/src/docs/userguide/dependencyManagementTerminology.adoc

```diff
@@ -0,0 +1,44 @@
+
+[[sub:terminology_version]]
+Version::
+
+A <<sub:terminology_module,module>> indicates its revision with the help of a version. Each module version contains a distinct set of changes. In practice there's no limitation to the scheme of the version. Timestamps, numbers, special suffixes like GA are all allowed identifiers. The most widely-used versioning strategy is link:https://semver.org/[semantic versioning]. Semantic versioning is represented by the scheme _major_._minor_._patch_ e.g. `1.4.15`. Each part conveys a special meaning about the backwards-compatibility of the release.
+
+[[sub:terminology_dependency]]
+Dependency::
+
+A dependency is defined as a versioned or unversioned module. Dependencies may provide <<sub:terminology_dependency_metadata,metadata>> to indicate that they depend on other dependencies required for it work properly, so-called <<sub:terminology_transitive_dependency,transitive dependencies>>. For more information, see <<declaring_dependencies>>. Gradle's dependency resolution process can apply constraints declared in the build script to select a specific version.
+
+[[sub:terminology_transitive_dependency]]
+Transitive dependency::
+
+A transitive dependency is a dependency required by another dependency to work properly. Dependencies hosted on a <<sec:terminology_repository,repository>> can provide <<sub:terminology_dependency_metadata,metadata>> declaring transitive dependencies. By default, Gradle resolves transitive dependencies if defined by metadata. The behavior is highly customizable. For more information, see <<managing_transitive_dependencies>>.
+
+[[sub:terminology_dependency_metadata]]
+Dependency metadata::
+
+Dependencies can provide metadata. Metadata is the data that describes the dependency in more detail e.g. the coordinates for locating it in a repository, information about the project or required <<sub:terminology_transitive_dependency,transitive dependencies>>. In Maven the metadata file is called `pom.xml`, in Ivy it is called `ivy.xml`.
+
+[[sub:terminology_configuration]]
+Configuration::
+
+A configuration represents the scope in a Gradle project a set of <<sub:terminology_dependency,dependencies>> should apply to e.g. the Guava dependency should be used to compile source code. The word "configuration" is an overloaded term and has a different meaning outside of the context of dependency management. For more information, see <<sub:scope_of_dependency_configurations>>.
+
+[[sub:terminology_repository]]
+Repository::
+
+A repository hosts a set of dependencies. The repository can be based on a binary repository product e.g. Artifactory or Nexus or a directory structure in the filesystem. Build scripts can define a set of repositories for resolving <<sub:terminology_dependency,dependencies>>. For more information, see <<declaring_repositories>>.
```

Comment:
"A repository hosts a set of dependencies." -> "A repository hosts a set of modules that can be used as dependencies." 

Link terms _moudle_ and _dependency_ to the corresponding sections in this Glossary.

Reply:
"...product e.g. Artifactory or Nexus or..." -> "...product (e.g. Artifactory or Nexus) or..."

Makes the sentence with the two _or_ easier to parse :).

Reply:
done and done

Reply:
A repository hosts a set of modules, each of which has a set of versions/releases/revisions/???. We really need a good term for this "module revision". But I don't think calling each of these a dependency is right.